x/cosmwasmpool/model/pool.go

```diff
@@ -0,0 +1,127 @@
+package model
+
+import (
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/cosmwasm/msg"
+	"github.com/osmosis-labs/osmosis/v15/x/cosmwasmpool/types"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+
+	cosmwasmutils "github.com/osmosis-labs/osmosis/osmoutils/cosmwasm"
+)
+
+type Pool struct {
+	CosmWasmPool
+	WasmKeeper types.WasmKeeper
+}
+
+var (
+	_ poolmanagertypes.PoolI  = &Pool{}
+	_ types.CosmWasmExtension = &Pool{}
+)
+
+// NewCosmWasmPool creates a new CosmWasm pool with the specified parameters.
+func NewCosmWasmPool(poolId uint64, codeId uint64, instantiateMsg []byte) Pool {
+	pool := Pool{
+		CosmWasmPool: CosmWasmPool{
+			PoolAddress:     poolmanagertypes.NewPoolAddress(poolId).String(),
+			ContractAddress: "", // N.B. This is to be set in InitializePool()
```

Comment:
should this be validated somewhere? maybe only return active=true if the contract is set? (to be fair, this will probably change on https://github.com/osmosis-labs/osmosis/issues/5328)

Reply:
This is the way it is because in `x/poolmanager` the execution flow is as follows:
1. Create poool: https://github.com/osmosis-labs/osmosis/blob/3ea287152994d2d1d05d81b302def529cf9054e4/x/poolmanager/create_pool.go#L112
- Here, we do not have contract address set yet
2. `InitializePool`: https://github.com/osmosis-labs/osmosis/blob/3ea287152994d2d1d05d81b302def529cf9054e4/x/poolmanager/create_pool.go#L133
- This is where the contract address gets set at contract instantiation time

Reply:
If any concerns, let's discuss in future PRs

Reply:
that makes sense. My thought was more adding a check that the pool has properly been on all methods, but I see that that makes the interface worse as they would have to return an error